extensions/wikia/PortableInfobox/tests/PortableInfoboxMobileRenderServiceTest.php

```diff
@@ -0,0 +1,637 @@
+								</div>
+								<div class="pi-item pi-data pi-item-spacing pi-border-color">
+									<h3 class="pi-data-label pi-secondary-font">test label</h3>
+									<div class="pi-data-value pi-font">test value</div>
+									</div>
+							</aside>',
+				'description' => 'Simple infobox with title, image and key-value pair',
+				'mockParams' => [
+					'extendImageData' => [
+						'alt' => 'image alt',
+						'url' => 'http://image.jpg',
+						'name' => 'test1',
+						'key' => 'test1',
+						'ref' => 1,
+						'width' => '400',
+						'height' => '200',
+						'thumbnail' => 'http://image.jpg',
+						'thumbnail2x' => 'http://image2x.jpg',
+						'media-type' => 'image',
+						'isVideo' => false
+					]
+				]
+			],
+			[
+				'input' => [
+					[
+						'type' => 'title',
+						'data' => [
+							'value' => 'Test Title'
+						]
+					],
+					[
+						'type' => 'image',
+						'data' => [ ]
+					],
+					[
+						'type' => 'data',
+						'data' => [
+							'label' => 'test label',
+							'value' => 'test value'
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<h2 class="pi-item pi-item-spacing pi-title">Test Title</h2>
+								<div class="pi-item pi-data pi-item-spacing pi-border-color">
+									<h3 class="pi-data-label pi-secondary-font">test label</h3>
+									<div class="pi-data-value pi-font">test value</div>
+									</div>
+							</aside>',
+				'description' => 'Simple infobox with title, INVALID image and key-value pair',
+				'mockParams' => []
+			],
+			[
+				'input' => [
+					[
+						'type' => 'title',
+						'data' => [
+							'value' => 'Test Title'
+						]
+					],
+					[
+						'type' => 'data',
+						'data' => [
+							'label' => 'test label',
+							'value' => 'test value'
+						]
+					]
+				],
+				'output' => '<aside class="portable-infobox pi-background">
+								<h2 class="pi-item pi-item-spacing pi-title">Test Title</h2>
+								<div class="pi-item pi-data pi-item-spacing pi-border-color">
+									<h3 class="pi-data-label pi-secondary-font">test label</h3>
+									<div class="pi-data-value pi-font">test value</div>
+								</div>
+							</aside>',
+				'description' => 'Simple infobox with title, empty image and key-value pair',
+				'mockParams' => []
+			],
+			[
+				'input' => [
+					[
+						'type' => 'title',
+						'data' => [
+							'value' => 'Test Title'
+						]
+					],
+					[
+						'type' => 'group',
+						'data' => [
+							'value' => [
+								[
+									'type' => 'header',
+									'data' => [
+										'value' => 'Test Header'
+									]
+								],
+								[
+									'type' => 'data',
+									'data' => [
+										'label' => 'test label',
+										'value' => 'test value'
+									]
+								],
+								[
+									'type' => 'data',
+									'data' => [
+										'label' => 'test label',
+										'value' => 'test value'
+									]
+								]
+							],
+							'layout' => 'default',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "layout" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)